mobile/src/actions/settings.js

```diff
@@ -80,14 +81,11 @@ export const registerDevice = () => async (dispatch, getState) => {
   }
   dispatch(checkURL(getState().mobile.settings.serverUrl))
   initClient(getState().mobile.settings.serverUrl, onRegister(dispatch), device)
-  await cozy.client.authorize().then(async ({ client }) => {
+  await cozy.client.authorize().then(({ client }) => {
     dispatch(unrevokeClient())
     dispatch(setClient(client))
-    const options = {
-      onError: onError(dispatch, getState),
-      onComplete: refreshFolder(dispatch, getState)
-    }
-    cozy.client.offline.startRepeatedReplication('io.cozy.files', 15, options)
+    dispatch(setOffline(true))
```

Comment:
Dispatching a bunch of actions sequentially for a given action creator is an antipattern: these dispatches look a lot like setter methods. It'd be better to dispatch a single action with a richer payload that will modify the state in multiple reducers.